modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/idempotency/NuxeoIdempotentFilter.java

```diff
@@ -0,0 +1,192 @@
+            return false;
+        }
+        String key = request.getHeader(HEADER_KEY);
+        if (key == null) {
+            log.debug("No idempotent processing done: no {} header present", HEADER_KEY);
+            return false;
+        }
+        log.debug("Idempotent request key: {}", key);
+        KeyValueService kvs = Framework.getService(KeyValueService.class);
+        if (kvs == null) {
+            log.debug("KeyValueService not present");
+            return false;
+        }
+        KeyValueStore store = kvs.getKeyValueStore(getStoreName());
+        String status = store.getString(key + INFO_SUFFIX);
+        if (status == null) {
+            log.debug("Handle new request for key: {}", key);
+            long ttl = getTTL().toSeconds();
+            store.put(key + INFO_SUFFIX, INPROGRESS_MARKER, ttl);
+            try {
+                try (CopyingResponseWrapper wrapper = new CopyingResponseWrapper(DEFERRED_OUTPUT_STREAM_THRESHOLD,
+                        response)) {
+                    chain.doFilter(request, wrapper);
+                    wrapper.flushBuffer();
+                    byte[] content = wrapper.getCaptureAsBytes();
+                    if (content.length > MAX_CONTENT_SIZE) {
+                        log.debug(
+                                "Not storing response for key: {} (status: {}, size in bytes: {}), max content size exceeded: {}",
+                                key, wrapper.getStatus(), content.length, MAX_CONTENT_SIZE);
+                        return true;
+                    }
+                    store.put(key, content, ttl);
+                    store.put(key + INFO_SUFFIX, NuxeoIdempotentResponse.write(wrapper), ttl);
+                    log.debug("Stored response for key: {} (status: {}, size in bytes: {})", key, wrapper.getStatus(),
+                            content.length);
+                    return true;
+                }
+            } catch (IOException | ServletException e) {
+                response.setStatus(SC_INTERNAL_SERVER_ERROR);
+                throw e;
+            } finally {
+                if (response.getStatus() >= SC_BAD_REQUEST) {
+                    // error request: cleanup store
+                    store.put(key, (String) null);
+                    store.put(key + INFO_SUFFIX, (String) null);
+                    log.debug("Cleanup store: error for key: {}", key);
+                }
+            }
+        } else if (INPROGRESS_MARKER.equals(status)) {
+            // request already in progress -> conflict
+            // Don't call response.sendError, because it commits the response
+            // which prevents NuxeoExceptionFilter from returning a custom error page.
+            response.setStatus(SC_CONFLICT);
+            log.debug("Conflict response for key: {}", key);
+            return true;
+        } else {
+            try {
+                // request already done: return stored result
```

Comment:
Here because it was already processed: if we called the filter chain, the request would be processed again and potentially break idempotency (for Drive use case, the file would be imported again, and a new document would be created again)